dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -373,81 +363,74 @@ public Response checkIntegrity(@Context HttpServletRequest httpServletRequest, @
               return response("No Auth Token set for endpoint", true);
             }
-            FormDataMultiPart form = new FormDataMultiPart();
-            form.field("AUTH_TOKEN",authToken.get());
 
-            //Sending bundle to endpoint
-            String url = endpoint.toURL()+"/api/integrity/generateintegritydata/";
+            final FormDataMultiPart form = new FormDataMultiPart();
+            form.field("AUTH_TOKEN", authToken.get());
 
-            Response response = postWithEndpointState(
+            //Sending bundle to endpoint
+            final String url = endpoint.toURL() + "/api/integrity/generateintegritydata/";
+            final Response response = postWithEndpointState(
             	endpoint.getId(), url, MediaType.TEXT_PLAIN_TYPE, Entity.entity(form, form.getMediaType())
             );
 
-            if(response.getStatus() == HttpStatus.SC_OK) {
+            if (response.getStatus() == HttpStatus.SC_OK) {
                 final String integrityDataRequestID = response.readEntity(String.class);
-
-                Thread integrityDataRequestChecker = new Thread() {
-
+                final Thread integrityDataRequestChecker = new Thread() {
                     @CloseDBIfOpened
                     public void run(){
-
-                        FormDataMultiPart form = new FormDataMultiPart();
+                        final FormDataMultiPart form = new FormDataMultiPart();
                         form.field("AUTH_TOKEN",authToken.get());
                         form.field("REQUEST_ID",integrityDataRequestID);
-
-                        String url = endpoint.toURL()+"/api/integrity/getintegritydata/";
+                        final String url = endpoint.toURL() + "/api/integrity/getintegritydata/";
 
                         boolean processing = true;
-
                         while(processing) {
+                            final Response response = postWithEndpointState(
+                                    endpoint.getId(),
+                                    url,
+                                    new MediaType("application", "zip"),
+                                    Entity.entity(form, form.getMediaType()));
 
-                        	Response response = postWithEndpointState(
-                        		endpoint.getId(), url, new MediaType("application", "zip"), Entity.entity(form, form.getMediaType())
-                        	);
-
-                            if ( response.getStatus() == HttpStatus.SC_OK ) {
-
+                            if (response.getStatus() == HttpStatus.SC_OK) {
                                 processing = false;
 
-                                InputStream zipFile = response.readEntity(InputStream.class);
-                                String outputDir = ConfigUtils.getIntegrityPath() + File.separator + endpoint.getId();
+                                final InputStream zipFile = response.readEntity(InputStream.class);
```

Comment:
Codacy found an issue: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)